Unit test generated by RoostGPT

roostAI opened the pull request:
```
Please pull these awesome changes in!
Test Name - roostAI-csharp-sample
AI Type   - Azure Open AI
AI Model  - roost-gpt4-32k
Test Type - Unit test
Use Type  - ui
Test Framework - nunit
```
Access Logs At - [https://dev.roost.ai/roostgpt/logs?trigger_id=72e02e60-e292-4c07-9dad-22232484dd43](https://dev.roost.ai/roostgpt/logs?trigger_id=72e02e60-e292-4c07-9dad-22232484dd43)

roostAI committed:
Unit test generated by RoostGPT

Using AI Model roost-gpt4-32k

roostAI committed:
Update CodeExamples.sln

Comment out the projects that have test build failing

mgdevstack commented on src/CodeExamples.sln:
```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "TestHashMap", "TestHashMap\TestHashMap.csproj", "{EBDC072A-4291-4482-9BAE-2490C190B1B6}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.CalculateUintCapacity", "HashMap.Test\HashMapNamespace.CalculateUintCapacity\HashMapNamespace.CalculateUintCapacity.csproj", "{4DC0AECB-322A-427A-B49B-FA54AF5A8D16}"
```
### 2/4 Tests Pass
```
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateUintCapacity (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateUintCapacity/HashMapNamespace.CalculateUintCapacity.csproj (in 476 ms).
  1 of 2 projects are up-to-date for restore.
  HashMap -> /workspaces/CodeExamples/src/HashMap/bin/Debug/net6.0/HashMap.dll
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateUintCapacity/CalculateUintCapacity.cs(96,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateUintCapacity/HashMapNamespace.CalculateUintCapacity.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateUintCapacity/CalculateUintCapacity.cs(111,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateUintCapacity/HashMapNamespace.CalculateUintCapacity.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateUintCapacity/CalculateUintCapacity.cs(125,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateUintCapacity/HashMapNamespace.CalculateUintCapacity.csproj]
  HashMapNamespace.CalculateUintCapacity -> /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateUintCapacity/bin/Debug/net7.0/HashMapNamespace.CalculateUintCapacity.dll
Test run for /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateUintCapacity/bin/Debug/net7.0/HashMapNamespace.CalculateUintCapacity.dll (.NETCoreApp,Version=v7.0)
Microsoft (R) Test Execution Command Line Tool Version 17.7.2 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
  Failed TestCalculateUintCapacity_HigherInitCapacity [65 ms]
  Error Message:
     Expected: 267
  But was:  266

  Stack Trace:
     at HashMapNamespace.Test.CalculateUintCapacityTests.TestCalculateUintCapacity_HigherInitCapacity() in /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateUintCapacity/CalculateUintCapacity.cs:line 111

  Failed TestCalculateUintCapacity_LowerInitCapacity [3 ms]
  Error Message:
     Expected: 3
  But was:  2

  Stack Trace:
     at HashMapNamespace.Test.CalculateUintCapacityTests.TestCalculateUintCapacity_LowerInitCapacity() in /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateUintCapacity/CalculateUintCapacity.cs:line 96


Failed!  - Failed:     2, Passed:     2, Skipped:     0, Total:     4, Duration: 78 ms - HashMapNamespace.CalculateUintCapacity.dll (net7.0)
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateUintCapacity (roost-1709036480) $
```

roostAI replied:
These tests fail in expectation due to the behavior of SMath.Round()
It is a minor gap.

mgdevstack commented on src/CodeExamples.sln:
```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.CalculateUintCapacity", "HashMap.Test\HashMapNamespace.CalculateUintCapacity\HashMapNamespace.CalculateUintCapacity.csproj", "{4DC0AECB-322A-427A-B49B-FA54AF5A8D16}"
+EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.CalculateCapacity", "HashMap.Test\HashMapNamespace.CalculateCapacity\HashMapNamespace.CalculateCapacity.csproj", "{52CF5877-0FA0-482F-A9E8-137C1FF052EC}"
```
### Compilation Error
```
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/HashMapNamespace.CalculateCapacity.csproj (in 429 ms).
  1 of 2 projects are up-to-date for restore.
  HashMap -> /workspaces/CodeExamples/src/HashMap/bin/Debug/net6.0/HashMap.dll
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/CalculateCapacity.cs(84,26): error CS0305: Using the generic type 'HashHelpers<T>' requires 1 type arguments [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/HashMapNamespace.CalculateCapacity.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/CalculateCapacity.cs(100,26): error CS0305: Using the generic type 'HashHelpers<T>' requires 1 type arguments [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/HashMapNamespace.CalculateCapacity.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/CalculateCapacity.cs(115,71): error CS0305: Using the generic type 'HashHelpers<T>' requires 1 type arguments [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/HashMapNamespace.CalculateCapacity.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/CalculateCapacity.cs(130,26): error CS0305: Using the generic type 'HashHelpers<T>' requires 1 type arguments [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/HashMapNamespace.CalculateCapacity.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/CalculateCapacity.cs(87,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/HashMapNamespace.CalculateCapacity.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/CalculateCapacity.cs(103,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/HashMapNamespace.CalculateCapacity.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/CalculateCapacity.cs(88,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/HashMapNamespace.CalculateCapacity.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/CalculateCapacity.cs(104,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/HashMapNamespace.CalculateCapacity.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/CalculateCapacity.cs(133,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/HashMapNamespace.CalculateCapacity.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/CalculateCapacity.cs(134,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity/HashMapNamespace.CalculateCapacity.csproj]
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.CalculateCapacity (roost-1709036480) $ 
```

mgdevstack commented on src/CodeExamples.sln:
```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
+EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.CalculateCapacity", "HashMap.Test\HashMapNamespace.CalculateCapacity\HashMapNamespace.CalculateCapacity.csproj", "{52CF5877-0FA0-482F-A9E8-137C1FF052EC}"
+# EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.ProvideCapacity", "HashMap.Test\HashMapNamespace.ProvideCapacity\HashMapNamespace.ProvideCapacity.csproj", "{6B3046FE-22DE-43B3-9335-83883D159BA1}"
```
### Class attributes are used as method causing compilation error
```
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.ProvideCapacity (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ProvideCapacity/HashMapNamespace.ProvideCapacity.csproj (in 443 ms).
  1 of 2 projects are up-to-date for restore.
  HashMap -> /workspaces/CodeExamples/src/HashMap/bin/Debug/net6.0/HashMap.dll
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ProvideCapacity/ProvideCapacity.cs(70,35): error CS1955: Non-invocable member 'HashMap64<int>.Capacity' cannot be used like a method. [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ProvideCapacity/HashMapNamespace.ProvideCapacity.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ProvideCapacity/ProvideCapacity.cs(72,46): error CS1955: Non-invocable member 'HashMap64<int>.Capacity' cannot be used like a method. [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ProvideCapacity/HashMapNamespace.ProvideCapacity.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ProvideCapacity/ProvideCapacity.cs(78,35): error CS1955: Non-invocable member 'HashMap64<int>.Capacity' cannot be used like a method. [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ProvideCapacity/HashMapNamespace.ProvideCapacity.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ProvideCapacity/ProvideCapacity.cs(83,28): error CS1955: Non-invocable member 'HashMap64<int>.Capacity' cannot be used like a method. [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ProvideCapacity/HashMapNamespace.ProvideCapacity.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ProvideCapacity/ProvideCapacity.cs(89,35): error CS1955: Non-invocable member 'HashMap64<int>.Capacity' cannot be used like a method. [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ProvideCapacity/HashMapNamespace.ProvideCapacity.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ProvideCapacity/ProvideCapacity.cs(92,40): error CS1955: Non-invocable member 'HashMap64<int>.Capacity' cannot be used like a method. [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ProvideCapacity/HashMapNamespace.ProvideCapacity.csproj]
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.ProvideCapacity (roost-1709036480) $ 
```

mgdevstack commented on src/CodeExamples.sln:
```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
+# EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.ProvideCapacity", "HashMap.Test\HashMapNamespace.ProvideCapacity\HashMapNamespace.ProvideCapacity.csproj", "{6B3046FE-22DE-43B3-9335-83883D159BA1}"
+# EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.Add", "HashMap.Test\HashMapNamespace.Add\HashMapNamespace.Add.csproj", "{8513B597-4E9C-44CD-809F-290F446E21E4}"
```
### 5/5 Tests Pass.
```
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.Add (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/HashMap/HashMap.csproj (in 7.18 sec).
  Restored /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Add/HashMapNamespace.Add.csproj (in 17.35 sec).
/workspaces/CodeExamples/src/HashMap/HashMap64.cs(214,21): warning CS8601: Possible null reference assignment. [/workspaces/CodeExamples/src/HashMap/HashMap.csproj]
/workspaces/CodeExamples/src/HashMap/HashMap64.cs(220,41): warning CS8604: Possible null reference argument for parameter 'value' in 'HashMap64Node.HashMap64Node(TValue value, long key, int next)'. [/workspaces/CodeExamples/src/HashMap/HashMap.csproj]
/workspaces/CodeExamples/src/HashMap/HashMap64.cs(235,41): warning CS8604: Possible null reference argument for parameter 'value' in 'HashMap64Node.HashMap64Node(TValue value, long key, int next)'. [/workspaces/CodeExamples/src/HashMap/HashMap.csproj]
/workspaces/CodeExamples/src/HashMap/HashMap64.cs(248,40): warning CS8604: Possible null reference argument for parameter 'value' in 'HashMap64Node.HashMap64Node(TValue value, long key, int next)'. [/workspaces/CodeExamples/src/HashMap/HashMap.csproj]
/workspaces/CodeExamples/src/HashMap/HashMap64.cs(295,12): warning CS8601: Possible null reference assignment. [/workspaces/CodeExamples/src/HashMap/HashMap.csproj]
/workspaces/CodeExamples/src/HashMap/HashMap64.cs(326,24): warning CS8601: Possible null reference assignment. [/workspaces/CodeExamples/src/HashMap/HashMap.csproj]
  HashMap -> /workspaces/CodeExamples/src/HashMap/bin/Debug/net6.0/HashMap.dll
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Add/Add.cs(139,28): warning CS8625: Cannot convert null literal to non-nullable reference type. [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Add/HashMapNamespace.Add.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Add/Add.cs(129,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Add/HashMapNamespace.Add.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Add/Add.cs(86,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Add/HashMapNamespace.Add.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Add/Add.cs(100,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Add/HashMapNamespace.Add.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Add/Add.cs(101,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Add/HashMapNamespace.Add.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Add/Add.cs(142,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Add/HashMapNamespace.Add.csproj]
  HashMapNamespace.Add -> /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Add/bin/Debug/net7.0/HashMapNamespace.Add.dll
Test run for /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Add/bin/Debug/net7.0/HashMapNamespace.Add.dll (.NETCoreApp,Version=v7.0)
Microsoft (R) Test Execution Command Line Tool Version 17.7.2 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.

Passed!  - Failed:     0, Passed:     5, Skipped:     0, Total:     5, Duration: 54 ms - HashMapNamespace.Add.dll (net7.0)
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.Add (roost-1709036480) $ 
```

mgdevstack commented on src/CodeExamples.sln:
```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
+# EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.Add", "HashMap.Test\HashMapNamespace.Add\HashMapNamespace.Add.csproj", "{8513B597-4E9C-44CD-809F-290F446E21E4}"
+EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.AddOrUpdate", "HashMap.Test\HashMapNamespace.AddOrUpdate\HashMapNamespace.AddOrUpdate.csproj", "{DD451542-3BE6-47B2-8D29-B614A0EE2F6B}"
```
### Compilation Error
```
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/HashMapNamespace.AddOrUpdate.csproj (in 428 ms).
  1 of 2 projects are up-to-date for restore.
  HashMap -> /workspaces/CodeExamples/src/HashMap/bin/Debug/net6.0/HashMap.dll
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/AddOrUpdate.cs(100,36): error CS1501: No overload for method 'AddOrUpdate' takes 3 arguments [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/HashMapNamespace.AddOrUpdate.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/AddOrUpdate.cs(108,83): warning CS8625: Cannot convert null literal to non-nullable reference type. [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/HashMapNamespace.AddOrUpdate.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/AddOrUpdate.cs(128,36): error CS1501: No overload for method 'AddOrUpdate' takes 3 arguments [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/HashMapNamespace.AddOrUpdate.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/AddOrUpdate.cs(102,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/HashMapNamespace.AddOrUpdate.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/AddOrUpdate.cs(102,39): warning NUnit2007: The actual value should not be a constant - perhaps the actual value and the expected value have switched places (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2007.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/HashMapNamespace.AddOrUpdate.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/AddOrUpdate.cs(120,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/HashMapNamespace.AddOrUpdate.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/AddOrUpdate.cs(120,45): warning NUnit2007: The actual value should not be a constant - perhaps the actual value and the expected value have switched places (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2007.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/HashMapNamespace.AddOrUpdate.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/AddOrUpdate.cs(130,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/HashMapNamespace.AddOrUpdate.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/AddOrUpdate.cs(130,39): warning NUnit2007: The actual value should not be a constant - perhaps the actual value and the expected value have switched places (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2007.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate/HashMapNamespace.AddOrUpdate.csproj]
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.AddOrUpdate (roost-1709036480) $ 
```

mgdevstack commented on src/CodeExamples.sln:
```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
+EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.AddOrUpdate", "HashMap.Test\HashMapNamespace.AddOrUpdate\HashMapNamespace.AddOrUpdate.csproj", "{DD451542-3BE6-47B2-8D29-B614A0EE2F6B}"
+# EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.Remove", "HashMap.Test\HashMapNamespace.Remove\HashMapNamespace.Remove.csproj", "{2FB8A7E4-D031-466B-8834-DA4DA8265808}"
```
### 5/5 Tests Pass.
```
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.Remove (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Remove/HashMapNamespace.Remove.csproj (in 475 ms).
  1 of 2 projects are up-to-date for restore.
  HashMap -> /workspaces/CodeExamples/src/HashMap/bin/Debug/net6.0/HashMap.dll
  HashMapNamespace.Remove -> /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Remove/bin/Debug/net7.0/HashMapNamespace.Remove.dll
Test run for /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Remove/bin/Debug/net7.0/HashMapNamespace.Remove.dll (.NETCoreApp,Version=v7.0)
Microsoft (R) Test Execution Command Line Tool Version 17.7.2 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.

Passed!  - Failed:     0, Passed:     5, Skipped:     0, Total:     5, Duration: 39 ms - HashMapNamespace.Remove.dll (net7.0)
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.Remove (roost-1709036480) $ 
```

mgdevstack commented on src/CodeExamples.sln:
```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
+# EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.Remove", "HashMap.Test\HashMapNamespace.Remove\HashMapNamespace.Remove.csproj", "{2FB8A7E4-D031-466B-8834-DA4DA8265808}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.ContainsKey", "HashMap.Test\HashMapNamespace.ContainsKey\HashMapNamespace.ContainsKey.csproj", "{49821606-F17E-4E4A-A1D0-5F5D052120F9}"
```
### 4/4 Tests Pass.
```
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.ContainsKey (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ContainsKey/HashMapNamespace.ContainsKey.csproj (in 480 ms).
  1 of 2 projects are up-to-date for restore.
  HashMap -> /workspaces/CodeExamples/src/HashMap/bin/Debug/net6.0/HashMap.dll
  HashMapNamespace.ContainsKey -> /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ContainsKey/bin/Debug/net7.0/HashMapNamespace.ContainsKey.dll
Test run for /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ContainsKey/bin/Debug/net7.0/HashMapNamespace.ContainsKey.dll (.NETCoreApp,Version=v7.0)
Microsoft (R) Test Execution Command Line Tool Version 17.7.2 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.

Passed!  - Failed:     0, Passed:     4, Skipped:     0, Total:     4, Duration: 20 ms - HashMapNamespace.ContainsKey.dll (net7.0)
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.ContainsKey (roost-1709036480) $ 
```

mgdevstack commented on src/CodeExamples.sln:
```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.ContainsKey", "HashMap.Test\HashMapNamespace.ContainsKey\HashMapNamespace.ContainsKey.csproj", "{49821606-F17E-4E4A-A1D0-5F5D052120F9}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.TryGetValue", "HashMap.Test\HashMapNamespace.TryGetValue\HashMapNamespace.TryGetValue.csproj", "{E252454D-7F25-45DC-9CF1-C6DB354EB67D}"
```
### 4/4 Tests Pass.
```
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.TryGetValue (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.TryGetValue/HashMapNamespace.TryGetValue.csproj (in 455 ms).
  1 of 2 projects are up-to-date for restore.
  HashMap -> /workspaces/CodeExamples/src/HashMap/bin/Debug/net6.0/HashMap.dll
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.TryGetValue/TryGetValue.cs(98,30): warning CS8600: Converting null literal or possible null value to non-nullable type. [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.TryGetValue/HashMapNamespace.TryGetValue.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.TryGetValue/TryGetValue.cs(100,85): warning CS8604: Possible null reference argument for parameter 'key' in 'bool Dictionary<string, int>.TryGetValue(string key, out int value)'. [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.TryGetValue/HashMapNamespace.TryGetValue.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.TryGetValue/TryGetValue.cs(80,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.TryGetValue/HashMapNamespace.TryGetValue.csproj]
  HashMapNamespace.TryGetValue -> /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.TryGetValue/bin/Debug/net7.0/HashMapNamespace.TryGetValue.dll
Test run for /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.TryGetValue/bin/Debug/net7.0/HashMapNamespace.TryGetValue.dll (.NETCoreApp,Version=v7.0)
Microsoft (R) Test Execution Command Line Tool Version 17.7.2 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.

Passed!  - Failed:     0, Passed:     4, Skipped:     0, Total:     4, Duration: 45 ms - HashMapNamespace.TryGetValue.dll (net7.0)
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.TryGetValue (roost-1709036480) $ 
```

mgdevstack commented on src/CodeExamples.sln:
```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.TryGetValue", "HashMap.Test\HashMapNamespace.TryGetValue\HashMapNamespace.TryGetValue.csproj", "{E252454D-7F25-45DC-9CF1-C6DB354EB67D}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.Clear", "HashMap.Test\HashMapNamespace.Clear\HashMapNamespace.Clear.csproj", "{C54A2E51-5AAA-42BE-BF3D-C5BF7FD9C557}"
```
### 4/4 Tests Pass.
```
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.Clear (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Clear/HashMapNamespace.Clear.csproj (in 481 ms).
  1 of 2 projects are up-to-date for restore.
  HashMap -> /workspaces/CodeExamples/src/HashMap/bin/Debug/net6.0/HashMap.dll
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Clear/Clear.cs(80,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Clear/HashMapNamespace.Clear.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Clear/Clear.cs(83,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Clear/HashMapNamespace.Clear.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Clear/Clear.cs(95,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Clear/HashMapNamespace.Clear.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Clear/Clear.cs(110,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Clear/HashMapNamespace.Clear.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Clear/Clear.cs(116,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Clear/HashMapNamespace.Clear.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Clear/Clear.cs(132,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Clear/HashMapNamespace.Clear.csproj]
  HashMapNamespace.Clear -> /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Clear/bin/Debug/net7.0/HashMapNamespace.Clear.dll
Test run for /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.Clear/bin/Debug/net7.0/HashMapNamespace.Clear.dll (.NETCoreApp,Version=v7.0)
Microsoft (R) Test Execution Command Line Tool Version 17.7.2 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.

Passed!  - Failed:     0, Passed:     4, Skipped:     0, Total:     4, Duration: 45 ms - HashMapNamespace.Clear.dll (net7.0)
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.Clear (roost-1709036480) $ 
```

mgdevstack commented on src/CodeExamples.sln:
```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.Clear", "HashMap.Test\HashMapNamespace.Clear\HashMapNamespace.Clear.csproj", "{C54A2E51-5AAA-42BE-BF3D-C5BF7FD9C557}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.GetEnumerator", "HashMap.Test\HashMapNamespace.GetEnumerator\HashMapNamespace.GetEnumerator.csproj", "{CB575984-6982-4487-8944-B4751E0970F0}"
```
### 4/4 Tests Pass.
```
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.GetEnumerator (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.GetEnumerator/HashMapNamespace.GetEnumerator.csproj (in 450 ms).
  1 of 2 projects are up-to-date for restore.
  HashMap -> /workspaces/CodeExamples/src/HashMap/bin/Debug/net6.0/HashMap.dll
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.GetEnumerator/GetEnumerator.cs(138,36): warning CS8625: Cannot convert null literal to non-nullable reference type. [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.GetEnumerator/HashMapNamespace.GetEnumerator.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.GetEnumerator/GetEnumerator.cs(140,37): warning CS8625: Cannot convert null literal to non-nullable reference type. [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.GetEnumerator/HashMapNamespace.GetEnumerator.csproj]
  HashMapNamespace.GetEnumerator -> /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.GetEnumerator/bin/Debug/net7.0/HashMapNamespace.GetEnumerator.dll
Test run for /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.GetEnumerator/bin/Debug/net7.0/HashMapNamespace.GetEnumerator.dll (.NETCoreApp,Version=v7.0)
Microsoft (R) Test Execution Command Line Tool Version 17.7.2 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.

Passed!  - Failed:     0, Passed:     4, Skipped:     0, Total:     4, Duration: 53 ms - HashMapNamespace.GetEnumerator.dll (net7.0)
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.GetEnumerator (roost-1709036480) $ 
```

mgdevstack commented on src/CodeExamples.sln:
```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.GetEnumerator", "HashMap.Test\HashMapNamespace.GetEnumerator\HashMapNamespace.GetEnumerator.csproj", "{CB575984-6982-4487-8944-B4751E0970F0}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.HashMap64Node", "HashMap.Test\HashMapNamespace.HashMap64Node\HashMapNamespace.HashMap64Node.csproj", "{1ABBB916-2C9C-4558-95EE-EDDCEBB4A8A1}"
```
### 5/5 Tests Pass.
```
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.HashMap64Node (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.HashMap64Node/HashMapNamespace.HashMap64Node.csproj (in 433 ms).
  1 of 2 projects are up-to-date for restore.
  HashMap -> /workspaces/CodeExamples/src/HashMap/bin/Debug/net6.0/HashMap.dll
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.HashMap64Node/HashMap64Node.cs(89,20): warning CS8618: Non-nullable property 'Value' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.HashMap64Node/HashMapNamespace.HashMap64Node.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.HashMap64Node/HashMap64Node.cs(158,21): warning CS8625: Cannot convert null literal to non-nullable reference type. [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.HashMap64Node/HashMapNamespace.HashMap64Node.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.HashMap64Node/HashMap64Node.cs(125,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.HashMap64Node/HashMapNamespace.HashMap64Node.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.HashMap64Node/HashMap64Node.cs(133,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.HashMap64Node/HashMapNamespace.HashMap64Node.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.HashMap64Node/HashMap64Node.cs(141,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.HashMap64Node/HashMapNamespace.HashMap64Node.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.HashMap64Node/HashMap64Node.cs(152,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.HashMap64Node/HashMapNamespace.HashMap64Node.csproj]
  HashMapNamespace.HashMap64Node -> /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.HashMap64Node/bin/Debug/net7.0/HashMapNamespace.HashMap64Node.dll
Test run for /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.HashMap64Node/bin/Debug/net7.0/HashMapNamespace.HashMap64Node.dll (.NETCoreApp,Version=v7.0)
Microsoft (R) Test Execution Command Line Tool Version 17.7.2 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.

Passed!  - Failed:     0, Passed:     5, Skipped:     0, Total:     5, Duration: 39 ms - HashMapNamespace.HashMap64Node.dll (net7.0)
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.HashMap64Node (roost-1709036480) $ 
```

mgdevstack commented on src/CodeExamples.sln:
```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.HashMap64Node", "HashMap.Test\HashMapNamespace.HashMap64Node\HashMapNamespace.HashMap64Node.csproj", "{1ABBB916-2C9C-4558-95EE-EDDCEBB4A8A1}"
+EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.ToString", "HashMap.Test\HashMapNamespace.ToString\HashMapNamespace.ToString.csproj", "{393CAC49-92C5-44D5-9019-96179864A0FF}"
```
### Compilation Error
```
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.ToString (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ToString/HashMapNamespace.ToString.csproj (in 420 ms).
  1 of 2 projects are up-to-date for restore.
  HashMap -> /workspaces/CodeExamples/src/HashMap/bin/Debug/net6.0/HashMap.dll
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ToString/ToString.cs(67,31): error CS0246: The type or namespace name 'HashMap<,>' could not be found (are you missing a using directive or an assembly reference?) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ToString/HashMapNamespace.ToString.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ToString/ToString.cs(83,31): error CS0246: The type or namespace name 'HashMap<,>' could not be found (are you missing a using directive or an assembly reference?) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ToString/HashMapNamespace.ToString.csproj]
/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ToString/ToString.cs(96,31): error CS0246: The type or namespace name 'HashMap<,>' could not be found (are you missing a using directive or an assembly reference?) [/workspaces/CodeExamples/src/HashMap.Test/HashMapNamespace.ToString/HashMapNamespace.ToString.csproj]
vscode ➜ .../CodeExamples/src/HashMap.Test/HashMapNamespace.ToString (roost-1709036480) $ 
```

mgdevstack commented on src/CodeExamples.sln:
```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
+EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMapNamespace.ToString", "HashMap.Test\HashMapNamespace.ToString\HashMapNamespace.ToString.csproj", "{393CAC49-92C5-44D5-9019-96179864A0FF}"
+# EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "RealtimePlugin.Run", "RealtimePlugin.Test\RealtimePlugin.Run\RealtimePlugin.Run.csproj", "{75DCAF84-C05A-4CFD-A0EF-A03426D009D4}"
```
### Compilation Error
```
vscode ➜ .../CodeExamples/src/RealtimePlugin.Test/RealtimePlugin.Run (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/PluginInterface/PluginInterface.csproj (in 2.03 sec).
  Restored /workspaces/CodeExamples/src/RealtimePlugin/RealtimePlugin.csproj (in 2.03 sec).
  Restored /workspaces/CodeExamples/src/RealtimePlugin.Test/RealtimePlugin.Run/RealtimePlugin.Run.csproj (in 2.04 sec).
  PluginInterface -> /workspaces/CodeExamples/src/PluginInterface/bin/Debug/net6.0/PluginInterface.dll
  RealtimePlugin -> /workspaces/CodeExamples/src/Plugins/Debug/net6.0/RealtimePlugin.dll
/workspaces/CodeExamples/src/RealtimePlugin.Test/RealtimePlugin.Run/Run.cs(97,57): error CS0246: The type or namespace name 'SimpleIocAdapter' could not be found (are you missing a using directive or an assembly reference?) [/workspaces/CodeExamples/src/RealtimePlugin.Test/RealtimePlugin.Run/RealtimePlugin.Run.csproj]
/workspaces/CodeExamples/src/RealtimePlugin.Test/RealtimePlugin.Run/Run.cs(104,65): error CS0117: 'State' does not contain a definition for 'Unavailable' [/workspaces/CodeExamples/src/RealtimePlugin.Test/RealtimePlugin.Run/RealtimePlugin.Run.csproj]
/workspaces/CodeExamples/src/RealtimePlugin.Test/RealtimePlugin.Run/Run.cs(116,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/RealtimePlugin.Test/RealtimePlugin.Run/RealtimePlugin.Run.csproj]
/workspaces/CodeExamples/src/RealtimePlugin.Test/RealtimePlugin.Run/Run.cs(126,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/RealtimePlugin.Test/RealtimePlugin.Run/RealtimePlugin.Run.csproj]
/workspaces/CodeExamples/src/RealtimePlugin.Test/RealtimePlugin.Run/Run.cs(136,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/RealtimePlugin.Test/RealtimePlugin.Run/RealtimePlugin.Run.csproj]
/workspaces/CodeExamples/src/RealtimePlugin.Test/RealtimePlugin.Run/Run.cs(159,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/RealtimePlugin.Test/RealtimePlugin.Run/RealtimePlugin.Run.csproj]
/workspaces/CodeExamples/src/RealtimePlugin.Test/RealtimePlugin.Run/Run.cs(149,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/RealtimePlugin.Test/RealtimePlugin.Run/RealtimePlugin.Run.csproj]
vscode ➜ .../CodeExamples/src/RealtimePlugin.Test/RealtimePlugin.Run (roost-1709036480) $ 
```

mgdevstack commented on src/CodeExamples.sln:
```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
+# EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "RealtimePlugin.Run", "RealtimePlugin.Test\RealtimePlugin.Run\RealtimePlugin.Run.csproj", "{75DCAF84-C05A-4CFD-A0EF-A03426D009D4}"
+# EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SimpleClassLibrary.ComposePlugins", "SimpleClassLibrary.Test\SimpleClassLibrary.ComposePlugins\SimpleClassLibrary.ComposePlugins.csproj", "{EF2FF9B2-CE63-4261-B1B3-931D8491428E}"
```
### 2/5 Tests Pass.
```
vscode ➜ .../CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/SimpleClassLibrary/SimpleClassLibrary.csproj (in 417 ms).
  Restored /workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/SimpleClassLibrary.ComposePlugins.csproj (in 456 ms).
  1 of 3 projects are up-to-date for restore.
  PluginInterface -> /workspaces/CodeExamples/src/PluginInterface/bin/Debug/net6.0/PluginInterface.dll
/workspaces/CodeExamples/src/SimpleClassLibrary/Composer.cs(12,55): warning CS8618: Non-nullable field 'plugins' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/workspaces/CodeExamples/src/SimpleClassLibrary/SimpleClassLibrary.csproj]
/workspaces/CodeExamples/src/SimpleClassLibrary/PluginLoader.cs(13,55): warning CS8618: Non-nullable field 'plugins' must contain a non-null value when exiting constructor. Consider declaring the field as nullable. [/workspaces/CodeExamples/src/SimpleClassLibrary/SimpleClassLibrary.csproj]
/workspaces/CodeExamples/src/SimpleClassLibrary/PluginLoader.cs(15,17): warning CS8618: Non-nullable property 'PluginLocation' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [/workspaces/CodeExamples/src/SimpleClassLibrary/SimpleClassLibrary.csproj]
/workspaces/CodeExamples/src/SimpleClassLibrary/PluginLoader.cs(16,20): warning CS8618: Non-nullable property 'Composer' must contain a non-null value when exiting constructor. Consider declaring the property as nullable. [/workspaces/CodeExamples/src/SimpleClassLibrary/SimpleClassLibrary.csproj]
  SimpleClassLibrary -> /workspaces/CodeExamples/src/SimpleClassLibrary/bin/Debug/net6.0/SimpleClassLibrary.dll
/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/ComposePlugins.cs(116,62): warning CS8600: Converting null literal or possible null value to non-nullable type. [/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/SimpleClassLibrary.ComposePlugins.csproj]
/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/ComposePlugins.cs(116,62): warning CS8625: Cannot convert null literal to non-nullable reference type. [/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/SimpleClassLibrary.ComposePlugins.csproj]
/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/ComposePlugins.cs(124,80): warning CS8625: Cannot convert null literal to non-nullable reference type. [/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/SimpleClassLibrary.ComposePlugins.csproj]
/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/ComposePlugins.cs(134,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/SimpleClassLibrary.ComposePlugins.csproj]
  SimpleClassLibrary.ComposePlugins -> /workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/bin/Debug/net7.0/SimpleClassLibrary.ComposePlugins.dll
Test run for /workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/bin/Debug/net7.0/SimpleClassLibrary.ComposePlugins.dll (.NETCoreApp,Version=v7.0)
Microsoft (R) Test Execution Command Line Tool Version 17.7.2 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
  Failed TestCompositionOfExistingPlugins [127 ms]
  Error Message:
   System.IO.DirectoryNotFoundException : Could not find a part of the path '/path/to/valid/plugin/directory'.
  Stack Trace:
     at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path, Boolean ignoreNotFound)
   at System.IO.Enumeration.FileSystemEnumerator`1.Init()
   at System.IO.Enumeration.FileSystemEnumerable`1..ctor(String directory, FindTransform transform, EnumerationOptions options, Boolean isNormalized)
   at System.IO.Enumeration.FileSystemEnumerableFactory.UserFiles(String directory, String expression, EnumerationOptions options)
   at System.IO.Directory.InternalEnumeratePaths(String path, String searchPattern, SearchTarget searchTarget, EnumerationOptions options)
   at System.IO.Directory.GetFiles(String path, String searchPattern, EnumerationOptions enumerationOptions)
   at System.ComponentModel.Composition.Hosting.DirectoryCatalog.GetFiles()
   at System.ComponentModel.Composition.Hosting.DirectoryCatalog.Initialize(String path, String searchPattern)
   at System.ComponentModel.Composition.Hosting.DirectoryCatalog..ctor(String path, String searchPattern)
   at System.ComponentModel.Composition.Hosting.DirectoryCatalog..ctor(String path)
   at SimpleClassLibrary.Composer.ComposePlugins(IPluginLoader pluginLoader) in /workspaces/CodeExamples/src/SimpleClassLibrary/Composer.cs:line 16
   at SimpleClassLibrary.Test.ComposePluginsTests.TestCompositionOfExistingPlugins() in /workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/ComposePlugins.cs:line 132
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
  Failed TestNullPluginLoader [47 ms]
  Error Message:
     Expected: <System.ArgumentNullException>
  But was:  <System.NullReferenceException: Object reference not set to an instance of an object.
   at SimpleClassLibrary.Composer.ComposePlugins(IPluginLoader pluginLoader) in /workspaces/CodeExamples/src/SimpleClassLibrary/Composer.cs:line 16
   at SimpleClassLibrary.Test.ComposePluginsTests.<>c__DisplayClass5_0.<TestNullPluginLoader>b__0() in /workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/ComposePlugins.cs:line 124
   at NUnit.Framework.Assert.Throws(IResolveConstraint expression, TestDelegate code, String message, Object[] args)>

  Stack Trace:
     at SimpleClassLibrary.Test.ComposePluginsTests.TestNullPluginLoader() in /workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/ComposePlugins.cs:line 124

  Failed TestSuccessfulPluginComposition [1 ms]
  Error Message:
   System.IO.DirectoryNotFoundException : Could not find a part of the path '/path/to/valid/plugin/directory'.
  Stack Trace:
     at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path, Boolean ignoreNotFound)
   at System.IO.Enumeration.FileSystemEnumerator`1.Init()
   at System.IO.Enumeration.FileSystemEnumerable`1..ctor(String directory, FindTransform transform, EnumerationOptions options, Boolean isNormalized)
   at System.IO.Enumeration.FileSystemEnumerableFactory.UserFiles(String directory, String expression, EnumerationOptions options)
   at System.IO.Directory.InternalEnumeratePaths(String path, String searchPattern, SearchTarget searchTarget, EnumerationOptions options)
   at System.IO.Directory.GetFiles(String path, String searchPattern, EnumerationOptions enumerationOptions)
   at System.ComponentModel.Composition.Hosting.DirectoryCatalog.GetFiles()
   at System.ComponentModel.Composition.Hosting.DirectoryCatalog.Initialize(String path, String searchPattern)
   at System.ComponentModel.Composition.Hosting.DirectoryCatalog..ctor(String path, String searchPattern)
   at System.ComponentModel.Composition.Hosting.DirectoryCatalog..ctor(String path)
   at SimpleClassLibrary.Composer.ComposePlugins(IPluginLoader pluginLoader) in /workspaces/CodeExamples/src/SimpleClassLibrary/Composer.cs:line 16
   at SimpleClassLibrary.Test.ComposePluginsTests.TestSuccessfulPluginComposition() in /workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins/ComposePlugins.cs:line 99
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)

Failed!  - Failed:     3, Passed:     2, Skipped:     0, Total:     5, Duration: 198 ms - SimpleClassLibrary.ComposePlugins.dll (net7.0)
vscode ➜ .../CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.ComposePlugins (roost-1709036480) $ 
```

mgdevstack commented on src/CodeExamples.sln:
```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
+# EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SimpleClassLibrary.ComposePlugins", "SimpleClassLibrary.Test\SimpleClassLibrary.ComposePlugins\SimpleClassLibrary.ComposePlugins.csproj", "{EF2FF9B2-CE63-4261-B1B3-931D8491428E}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SimpleClassLibrary.LoadAndRunPlugins", "SimpleClassLibrary.Test\SimpleClassLibrary.LoadAndRunPlugins\SimpleClassLibrary.LoadAndRunPlugins.csproj", "{0B70D31A-0736-434E-A7E3-3A460F050CB3}"
```
### 5/5 Tests Pass.
```
vscode ➜ .../CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.LoadAndRunPlugins (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.LoadAndRunPlugins/SimpleClassLibrary.LoadAndRunPlugins.csproj (in 466 ms).
  2 of 3 projects are up-to-date for restore.
  PluginInterface -> /workspaces/CodeExamples/src/PluginInterface/bin/Debug/net6.0/PluginInterface.dll
  SimpleClassLibrary -> /workspaces/CodeExamples/src/SimpleClassLibrary/bin/Debug/net6.0/SimpleClassLibrary.dll
/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.LoadAndRunPlugins/LoadAndRunPlugins.cs(117,58): warning CS8625: Cannot convert null literal to non-nullable reference type. [/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.LoadAndRunPlugins/SimpleClassLibrary.LoadAndRunPlugins.csproj]
/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.LoadAndRunPlugins/LoadAndRunPlugins.cs(110,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.LoadAndRunPlugins/SimpleClassLibrary.LoadAndRunPlugins.csproj]
/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.LoadAndRunPlugins/LoadAndRunPlugins.cs(120,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.LoadAndRunPlugins/SimpleClassLibrary.LoadAndRunPlugins.csproj]
/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.LoadAndRunPlugins/LoadAndRunPlugins.cs(134,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.LoadAndRunPlugins/SimpleClassLibrary.LoadAndRunPlugins.csproj]
/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.LoadAndRunPlugins/LoadAndRunPlugins.cs(148,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.LoadAndRunPlugins/SimpleClassLibrary.LoadAndRunPlugins.csproj]
/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.LoadAndRunPlugins/LoadAndRunPlugins.cs(169,13): warning NUnit2005: Consider using the constraint model, Assert.That(actual, Is.EqualTo(expected)), instead of the classic model, Assert.AreEqual(expected, actual) (https://github.com/nunit/nunit.analyzers/tree/master/documentation/NUnit2005.md) [/workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.LoadAndRunPlugins/SimpleClassLibrary.LoadAndRunPlugins.csproj]
  SimpleClassLibrary.LoadAndRunPlugins -> /workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.LoadAndRunPlugins/bin/Debug/net7.0/SimpleClassLibrary.LoadAndRunPlugins.dll
Test run for /workspaces/CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.LoadAndRunPlugins/bin/Debug/net7.0/SimpleClassLibrary.LoadAndRunPlugins.dll (.NETCoreApp,Version=v7.0)
Microsoft (R) Test Execution Command Line Tool Version 17.7.2 (x64)
Copyright (c) Microsoft Corporation.  All rights reserved.

Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
No plugins with type: validType

Loaded plugin: 
Plugin returned NoError


Passed!  - Failed:     0, Passed:     5, Skipped:     0, Total:     5, Duration: 177 ms - SimpleClassLibrary.LoadAndRunPlugins.dll (net7.0)
vscode ➜ .../CodeExamples/src/SimpleClassLibrary.Test/SimpleClassLibrary.LoadAndRunPlugins (roost-1709036480) $ 
```

mgdevstack commented on src/CodeExamples.sln:
```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SimpleClassLibrary.LoadAndRunPlugins", "SimpleClassLibrary.Test\SimpleClassLibrary.LoadAndRunPlugins\SimpleClassLibrary.LoadAndRunPlugins.csproj", "{0B70D31A-0736-434E-A7E3-3A460F050CB3}"
+EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SimpleConsoleApp.typeof", "SimpleConsoleApp.Test\SimpleConsoleApp.typeof\SimpleConsoleApp.typeof.csproj", "{A578A014-9E4E-4033-A67D-37D912AAAEB9}"
```
### Compilation Error
```
vscode ➜ .../CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/SimpleConsoleApp.typeof.csproj (in 641 ms).
  Restored /workspaces/CodeExamples/src/SimpleConsoleApp/SimpleConsoleApp.csproj (in 2.73 sec).
  2 of 4 projects are up-to-date for restore.
  PluginInterface -> /workspaces/CodeExamples/src/PluginInterface/bin/Debug/net6.0/PluginInterface.dll
  SimpleClassLibrary -> /workspaces/CodeExamples/src/SimpleClassLibrary/bin/Debug/net6.0/SimpleClassLibrary.dll
/workspaces/CodeExamples/src/SimpleConsoleApp/SimpleServiceLocator.cs(19,26): warning CS8601: Possible null reference assignment. [/workspaces/CodeExamples/src/SimpleConsoleApp/SimpleConsoleApp.csproj]
/workspaces/CodeExamples/src/SimpleConsoleApp/SimpleServiceLocator.cs(25,42): warning CS8600: Converting null literal or possible null value to non-nullable type. [/workspaces/CodeExamples/src/SimpleConsoleApp/SimpleConsoleApp.csproj]
/workspaces/CodeExamples/src/SimpleConsoleApp/SimpleServiceLocator.cs(26,11): warning CS8603: Possible null reference return. [/workspaces/CodeExamples/src/SimpleConsoleApp/SimpleConsoleApp.csproj]
/workspaces/CodeExamples/src/SimpleConsoleApp/SimpleServiceLocator.cs(31,57): warning CS8625: Cannot convert null literal to non-nullable reference type. [/workspaces/CodeExamples/src/SimpleConsoleApp/SimpleConsoleApp.csproj]
  SimpleConsoleApp -> /workspaces/CodeExamples/src/SimpleConsoleApp/bin/Debug/net6.0/SimpleConsoleApp.dll
/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/typeof.cs(83,40): error CS0246: The type or namespace name 'IService' could not be found (are you missing a using directive or an assembly reference?) [/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/SimpleConsoleApp.typeof.csproj]
/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/typeof.cs(89,87): error CS0246: The type or namespace name 'IService' could not be found (are you missing a using directive or an assembly reference?) [/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/SimpleConsoleApp.typeof.csproj]
/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/typeof.cs(96,41): error CS0246: The type or namespace name 'IService' could not be found (are you missing a using directive or an assembly reference?) [/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/SimpleConsoleApp.typeof.csproj]
/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/typeof.cs(97,41): error CS0246: The type or namespace name 'IService' could not be found (are you missing a using directive or an assembly reference?) [/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/SimpleConsoleApp.typeof.csproj]
/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/typeof.cs(104,88): error CS0246: The type or namespace name 'IService' could not be found (are you missing a using directive or an assembly reference?) [/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/SimpleConsoleApp.typeof.csproj]
/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/typeof.cs(111,13): error CS0246: The type or namespace name 'IService' could not be found (are you missing a using directive or an assembly reference?) [/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/SimpleConsoleApp.typeof.csproj]
/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/typeof.cs(121,44): error CS0246: The type or namespace name 'IService' could not be found (are you missing a using directive or an assembly reference?) [/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/SimpleConsoleApp.typeof.csproj]
/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/typeof.cs(122,47): error CS0246: The type or namespace name 'IAnotherService' could not be found (are you missing a using directive or an assembly reference?) [/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/SimpleConsoleApp.typeof.csproj]
/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/typeof.cs(129,91): error CS0246: The type or namespace name 'IService' could not be found (are you missing a using directive or an assembly reference?) [/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/SimpleConsoleApp.typeof.csproj]
/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/typeof.cs(130,94): error CS0246: The type or namespace name 'IAnotherService' could not be found (are you missing a using directive or an assembly reference?) [/workspaces/CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof/SimpleConsoleApp.typeof.csproj]
vscode ➜ .../CodeExamples/src/SimpleConsoleApp.Test/SimpleConsoleApp.typeof (roost-1709036480) $ 
```

mgdevstack commented on src/CodeExamples.sln:
```diff
@@ -23,6 +23,40 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "HashMap", "HashMap\HashMap.
+EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SimpleConsoleApp.typeof", "SimpleConsoleApp.Test\SimpleConsoleApp.typeof\SimpleConsoleApp.typeof.csproj", "{A578A014-9E4E-4033-A67D-37D912AAAEB9}"
+# EndProject
+# Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "SimulationPlugin.Run", "SimulationPlugin.Test\SimulationPlugin.Run\SimulationPlugin.Run.csproj", "{9F774258-DD1F-49D7-B82B-0A59D3D7D594}"
```
### Compilation Error
```
vscode ➜ .../CodeExamples/src/SimulationPlugin.Test/SimulationPlugin.Run (roost-1709036480) $ dotnet test
  Determining projects to restore...
  Restored /workspaces/CodeExamples/src/SimulationPlugin/SimulationPlugin.csproj (in 419 ms).
  Restored /workspaces/CodeExamples/src/SimulationPlugin.Test/SimulationPlugin.Run/SimulationPlugin.Run.csproj (in 451 ms).
  1 of 3 projects are up-to-date for restore.
  PluginInterface -> /workspaces/CodeExamples/src/PluginInterface/bin/Debug/net6.0/PluginInterface.dll
  SimulationPlugin -> /workspaces/CodeExamples/src/Plugins/Debug/net6.0/SimulationPlugin.dll
/workspaces/CodeExamples/src/SimulationPlugin.Test/SimulationPlugin.Run/Run.cs(79,49): error CS0246: The type or namespace name 'UnityServiceLocator' could not be found (are you missing a using directive or an assembly reference?) [/workspaces/CodeExamples/src/SimulationPlugin.Test/SimulationPlugin.Run/SimulationPlugin.Run.csproj]
vscode ➜ .../CodeExamples/src/SimulationPlugin.Test/SimulationPlugin.Run (roost-1709036480) $ 
```

roostAI committed:
Improved Unit test generated by RoostGPT

Using AI Model roost-gpt4-32k, for user feedback: Test case must be validating by calling the Class and method under test rather than redeclaring and changing the method signature. Do not declare any class or method whose behaviour is under test. I expect you to improvise this CalculateUintCapacityTests by invoking method CalculateUintCapacity residing in class HashHelpers<T> in namespace HashMapNamespace. Prefer mocking using moq only if required to test static class.
